### DISCUSSION TITLE
I have a set of PNG files in the following sizes: 256x256, 128x128, 96x96, 64x64, 48x48, 32x32, 24x24, 20x20, 16x16. How can I convert all of these into a single windows .ICO file?

### DISCUSSION THREAD

**Discussion by futuremotiondev** [ImageMagick/ImageMagick#7196] [Help]
### ImageMagick version

7.1.1-27 Q16-HDRI x64 0225ac4:20240121

### Operating system, version and so on

Windows 10 Pro x64 22H2 (OS Build 19045.4170)

### Description

I want to properly convert a large number of PNGs with standard icon sizes to an ICO file using ImageMagick. I've googled heavily and haven't found any kind of answer. Nothing recent at least.

Can someone please point me in the right direction? Would should my command look like (I'm on windows 10)?

**Comment by fmw42** [comment]
I am not a Windows user nor an ico expert, but have you tried

`magick image.png image.ico`

**Reply by fmw42** [reply]
Also have you checked

`magick -list format`

Does it list ICO and is r or rw?  If only r, then you can only read ICO files. If rw, then you can write them also.

**Comment by futuremotiondev** [comment]
I got it to work. I'm using powershell, here's the code that did it:

```powershell

try {
    $MagickCMD = Get-Command magick.exe -CommandType Application
} catch {
    throw "Can't find magick.exe (ImageMagick) in PATH"
}

$ICOOutput = "C:\users\username\Desktop\NewIcon.ico"

# Populate $FinalIcons array
$FinalIcons = Get-ChildItem $TempDir -Filter "FinalIcon-*.png" | ForEach-Object { $_.Name }

# Create the icon
$ICOOutput = Get-UniqueFileOrFolderNameIfDuplicate -Path $ICOOutput
& $MagickCmd $($FinalIcons.ForEach{"$TempDir\$_"}) $ICOOutput | Out-Null
```

**Comment by futuremotiondev** [comment]
Closing since I figured it all out, thanks!
